test/middlewares/generateDocs.test.ts

```diff
@@ -0,0 +1,68 @@
+import 'jest-extended'
+import { tmpdir } from 'os'
+import fs from 'fs-extra'
+import generateDocs from '../../src/middlewares/generateDocs'
+import AsyncAPIDocument from '@asyncapi/parser/lib/models/asyncapi'
+
+const TEST_ASYNCAPI_DOCUMENT = new AsyncAPIDocument({
+  asyncapi: '2.2.0',
+  info: {
+    title: 'Account Service',
+    version: '1.0.0',
+    description: 'lorem ipsum'
+  },
+  servers: {
+    test: {
+      url: 'mqtt://fake-url',
+      protocol: 'mqtt',
+    },
+  },
+  channels: {
+    'test/channel': {
+      publish: {
+        message: {
+          payload: {
+            type: 'string',
+          },
+        },
+      },
+    },
+  },
+})
+
+const CONFIG_TEST_DATA = {
+  websocket: {
+    httpServer: 'customServer',
+    adapter: 'native', // Default. Can also be 'socket.io' or a reference to a custom adapter.
+  },
+  cluster: {
+    adapter: 'redis',
+    name: 'gleeCluster',
+    url: 'redis://localhost:6379',
+  },
+  generator: {
+    folder: 'output',
+    template: 'markdown-template',
+  },
+}
+
+describe('generateDocs', () => {
+  it('should generate documentation', async () => {
+    const testDir = tmpdir() + `/${CONFIG_TEST_DATA.generator.folder}`;
+    fs.emptyDirSync(testDir);
+    let err: Error | undefined = undefined;
```

Comment:
I think you can do something like this:

```suggestion
    let err?: Error
```

or like this, I can't remember:

```suggestion
    let err: Error?
```

Reply:
@fmvilas I don't think I understand what you're trying to do here but it seems your code snippet is wrong :) 

Mind explaining what you're trying to achieve with that snippet?

Reply:
I was trying to do something like this: https://www.typescriptlang.org/docs/handbook/advanced-types.html#optional-parameters-and-properties. But it looks like it only works on parameters and object properties 🤦

Reply:
In any case, you can simplify it like this:

```suggestion
    let err: Error | undefined
```